packages/context/src/context.ts

```diff
@@ -20,11 +24,37 @@ export class Context {
         throw new Error(`Cannot rebind key "${key}", associated binding is locked`);
     }
 
-    const binding = new Binding(key);
+    const binding = new Binding(this, key);
     this.registry.set(key, binding);
     return binding;
   }
 
+  createClassInstance<T>(ctor: Constructor<T>) : T {
+    const args = this._resolveInjectedArguments(ctor);
+    return new ctor(...args);
+  }
+
+  private _resolveInjectedArguments(fn: Function): BoundValue[] {
+    const args: BoundValue[] = [];
+    // NOTE: the array may be sparse, i.e.
+    //   Object.keys(injectedArgs).length !== injectedArgs.length
+    // Example value:
+    //   [ , 'key1', , 'key2']
+    const injectedArgs = describeInjectedArguments(fn);
+
+    for (let ix = 0; ix < fn.length; ix++) {
+      const bindingKey = injectedArgs[ix];
+      if (!bindingKey) {
```

Comment:
This will throw on the first argument that's not injected, ignoring any other arguments that may be missing `@inject` decorator too. As a result, the user may need more than one iteration to fix the code.

We can detect all undecorated arguments upfront, but is the added complexity worth the cost? Thoughts?